### PR TITLE
Devops/edit single recipe (PRA-126)

### PR DESCRIPTION
**Pull request by AnnaPestova1**
## One Line Description
Edit recipe logic
## Requirements
update backend (new endpoints and methods!), we cannot edit image yet
_What this code should do_
go to saved recipe - click on recipe card - on the page with single recipe click on edit button - try to edit recipe and save id (we do not edit image yet!)
## Notes

_Any additional info_

## Test Steps

_Instructions on how to test the changes_

## UI/UX Outcome

_Attach a screenshot here_
image

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed

### CODE REVIEW

**Review by AnnaSolovykh — APPROVED** [review]
LGTM!

File: src/components/EditRecipe.tsx [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+                    <b>Servings</b>
+                  </FormLabel>
+                  <InputGroup w="50%">
+                    <Input
+                      size="sm"
+                      value={recipe.recipeServings || ""}
+                      onChange={e => {
+                        setRecipe({
+                          ...recipe,
+                          recipeServings: Number(e.target.value)
```
Comment (Ekaterina-Bondareva) [inline src/components/EditRecipe.tsx:304]: Could you please add check that value is numeric before converting to Number?

File: src/components/EditRecipe.tsx [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+                      <FormLabel></FormLabel>
+                      <Input
+                        size="sm"
+                        name="ingredientAmount"
+                        type="number"
+                        value={ingredient.ingredientAmount || ""}
+                        onChange={e => {
+                          const newIngredients = [...ingredients];
+                          newIngredients[index].ingredientAmount = Number(
+                            e.target.value
```
Comment (Ekaterina-Bondareva) [inline src/components/EditRecipe.tsx:369]: Same check for numeric value here as well.

File: src/components/EditRecipe.tsx [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+                    <FormControl w="40%" marginRight="3">
+                      <FormLabel></FormLabel>
+                      <Input
+                        size="sm"
+                        name="ingredientAmount"
+                        type="number"
+                        value={ingredient.ingredientAmount || ""}
+                        onChange={e => {
+                          const newIngredients = [...ingredients];
+                          newIngredients[index].ingredientAmount = Number(
```
Comment (Ekaterina-Bondareva) [inline src/components/EditRecipe.tsx:368]: Probably a check should be added to not allow empty/zero value for ingredient amount. As of now it is possible to leave field empty and save recipe with 0 amount of an ingredient.

**Review by Ekaterina-Bondareva — CHANGES_REQUESTED** [review]
Removing tags doesn't work. It is possible to remove tag clicking on - button but when saving recipe tags are sent to backend unchanged.

**Comment by Ekaterina-Bondareva** [comment]
When clicking Save button PATRCH request is submitted twice to backend due to `saveRecipe` being added as onClick on Save button and as onSubmit on the whole form.
Also digits are removed from tag values  so '3 persons' will be saved as 'persons'. And if tag value is a number then backend will fail to save recipe at all and send error in response

File: src/components/EditRecipe.tsx (hunk 3 above)
Comment (AnnaPestova1) [reply src/components/EditRecipe.tsx:368]: I prohibit the oportunity to enter numbers less than 0 and handle the form rendering for numbers = 0. We discussed and desided to leave the oportinity fo user to enter just ingredients without amount

**Comment by AnnaPestova1** [comment]
> Removing tags doesn't work. It is possible to remove tag clicking on - button but when saving recipe tags are sent to backend unchanged.

Thanks! I missed to change setter

**Comment by AnnaPestova1** [comment]
> When clicking Save button PATCH request is submitted twice to backend due to `saveRecipe` being added as onClick on Save button and as onSubmit on the whole form.
> Also digits are removed from tag values so '3 persons' will be saved as 'persons'. And if tag value is a number then backend will fail to save recipe at all and send error in response (looks like a backend issue though).

Thanks! Done

**Review by Ekaterina-Bondareva — APPROVED** [review]
LGTM!

**Pull request merged**